cpp/memilio/math/interpolation.h

```diff
@@ -40,6 +41,38 @@ auto linear_interpolation(const X& x_eval, const X& x_1, const X& x_2, const V&
     return y1 + weight * (y2 - y1);
 }
 
+/**
+ * @brief Linear interpolation between two point of a dataset, which is represented by a vector of pairs of node and value.
+ * @param[in] vector Vector of pairs of node and value.
+ * @param[in] node The node whose value need to be found.
+ * @param[out] unnamed Interpolation result.
+ */
+template <typename X, typename Y>
+Y linear_interpolation_of_data_set(const std::vector<std::pair<X, Y>>& vector, const X& node)
+{
+    // If the vector is empty or has only 1 node, return 0
+    if (vector.empty() || vector.size() == 1) {

```

Comment:
Also, it might be a problem if we use it like this and the interpolation suddenly drops to 0 after the last data point. I know this is not interpolation then anymore, but we have to have this in mind when setting up immunity like this.

Reply:
I adjust it to throw a `std::invalid_argument`